example/sequence_container_tests.cpp

```diff
@@ -1,20 +1,67 @@
+#include <deque>
+#include <list>
+#include <set>
+
 #include "tests.h"
 
 using namespace snowhouse;
 
-template<typename T>
-void SequenceContainerActual()
+static const char* ExpectedActual = "\nActual: [ 1, 2, 3, 5, 8 ]";
+
+static void insert_numbers(std::multiset<int>& container)
 {
-  const char* ExpectedActual = "\nActual: [ 1, 2, 3, 5, 8 ]";
+  container.insert(1);
+  container.insert(2);
+  container.insert(3);
+  container.insert(5);
+  container.insert(8);
+}
 
-  T container;
-  container.clear();
+static void insert_numbers(std::set<int>& container)
+{
+  container.insert(1);
+  container.insert(2);
+  container.insert(3);
+  container.insert(5);
+  container.insert(8);
+}
+
+#if __cplusplus >= 201103L
+#include <array>
+#include <forward_list>
+
+static void insert_numbers(std::array<int, 5>& container)
+{
+  container[0] = 1;
+  container[1] = 2;
+  container[2] = 3;
+  container[3] = 5;
+  container[4] = 8;
+}
+
+static void insert_numbers(std::forward_list<int>& container)
+{
+  container.push_front(8);
+  container.push_front(5);
+  container.push_front(3);
+  container.push_front(2);
+  container.push_front(1);
+}
+#endif
+
+template<typename T>
+static void insert_numbers(T& container)
+{
   container.push_back(1);
   container.push_back(2);
   container.push_back(3);
   container.push_back(5);
   container.push_back(8);
+}
```

Comment:
The other functions should probably template specializations instead of non-template overloaded functions.